src/explore-education-statistics-common/src/modules/charts/components/LineChartBlock.tsx

```diff
@@ -104,17 +105,23 @@ const LineChartBlock = ({
 
   const minorDomainTicks = getMinorAxisDomainTicks(chartData, axes.minor);
   const majorDomainTicks = getMajorAxisDomainTicks(chartData, axes.major);
-
-  const yAxisWidth = parseNumber(axes.minor.size);
-  const xAxisHeight = parseNumber(axes.major.size);
-
   const dataSetCategoryConfigs = getDataSetCategoryConfigs(
     dataSetCategories,
     legend.items,
     meta,
   );
-
   const minorAxisDecimals = getMinorAxisDecimalPlaces(dataSetCategoryConfigs);
+  const minorAxisUnit = getUnitOrDefault(
+    axes.minor.unit,
+    dataSetCategoryConfigs,
+  );
```

Comment:
This function name is a little confusing. The name implies that it should try and find unit, and if it cannot, it should use a default. Looking at the code, this isn't the case though, as it just uses the first param (the 'default') if it's been set.

There's also something about the parameter order that doesn't sit right as it's allow undefined for the first value (typically optional/undefined properties should be set last in a function signature as these can be afforded default parameters).

I would suggest to improve the semantics of this, we refactor this to the following:

```ts
const minorAxisUnit = axes.minor.unit ?? getUnit(dataSetCategoryConfigs);
```

Where:

```ts
function getUnit(dataSetCategoryConfigs: DataSetCategoryConfig[]): string 
```

Note that I've included an explicit return type - this is good practice.